cloud/google/clients/computeservice_test.go

```diff
@@ -0,0 +1,121 @@
+package clients_test
+
+import (
+	compute "google.golang.org/api/compute/v1"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/api/googleapi"
+	"net/http"
+	"net/http/httptest"
+	"sigs.k8s.io/cluster-api/cloud/google/clients"
+	"testing"
+)
+
+func TestImagesGet(t *testing.T) {
+	mux, server, client := createMuxServerAndClient()
+	defer server.Close()
+	responseImage := compute.Image{
+		Name:             "imageName",
+		ArchiveSizeBytes: 544,
+	}
+	mux.Handle("/compute/v1/projects/projectName/global/images/imageName", handler(nil, &responseImage))
+	image, err := client.ImagesGet("projectName", "imageName")
+	assert.Nil(t, err)
```

Comment:
If you're going to use and assertion library, at least provide useful failure messages. This is hardly helpful:

```go
func TestFoo(t *testing.T) {                                                                                                                                                                              
    x := 1                                                                                                                                                                                                
    y := 2                                                                                                                                                                                                
    assert.Equal(t, x, y)                                                                                                                                                                                        
} 
```

```console
go test                                                                                                                                                                                                   
--- FAIL: TestFoo (0.00s)                                                                                                                                                                                 
        Error Trace:                                                                                                                                                                                      
        Error:          Not equal: 1 (expected)                                                                                                                                                           
                                != 2 (actual)
```

Now for a more opinionated comment, please don't use assertion libraries if you can. This a request, not a mandate. I would state my reasons, but the Go team sums them up pretty neatly: https://golang.org/doc/faq#testing_framework.

Reply:
I think the go testing library does support letting the test finish and then all failed assertions will print. I sort of understand the anti-assertion library sentiment. I think one good thing about using the library is that the error messages are consistent and easy to scan & read. However, as you pointed out, that's not very useful if the failure messages aren't populated in which case there is a bit of a custom message anyway.

I was looking through some of the k8s.io libraries and it seems that there is a mix of using the assert library and implementing the checks. I'll go with what you suggested and rework these tests to not u se assert since it seems that is somewhat the norm.